### PR TITLE
Admin login

### PR DESCRIPTION
**Pull request by KombangkoeDias**
Finished Admin Login both backend and frontend, 
Changes : 
1. As I now try to use the MacOS to develop, in server, I change bcrypt library to bcryptjs for more universal compatibility. 
2. Added AdminRouterGuard and tested.
3. One can add his admin user by doing a POST request to `serverURL/admin/create` with four body fields :
- email : '-your new admin email-'
- password : '-your new password-'
- userType: 'admin',
- secret: 'Kojira_secret_code'
4. the admin login page will be at `/admin/login`

### CODE REVIEW

**Comment by puttimeth** [comment]
The serverURL is still set to localserver. Don't forget to change it if it's not intention.

**Comment by KombangkoeDias** [comment]
> The serverURL is still set to localserver. Don't forget to change it if it's not intention.

Thanks, forgot about that.

**Comment by Megacodexz** [comment]
I got this error when starting the app.

Code_2021-03-28_20-04-50

File: nontclient/src/Router/mainRouter.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ import ReserveInfo from "../Components/Reservation/ReserveInfo";
 import GuardedRoute from "./GuardedRoute";
 import NontOwnerRouter from "./NontOwnerRouter";
 import NontSitterRouter from "./NontSitterRouter";
+import AdminLogin from "../Components/Admin/AdminLogin";
+import AdminCreate from "../Components/Admin/AdminCreate";
```
Comment (Megacodexz) [inline nontclient/src/Router/mainRouter.js:14]: It ran perfectly after I had deleted this line.

**Comment by Megacodexz** [comment]
I also encountered this warning in the console after I had logged in.

firefox_2021-03-28_20-17-19

I don't know whether it is severe or not, but it might be worth investigating.

File: nontclient/src/Router/mainRouter.js (hunk 1 above)
Comment (KombangkoeDias) [reply nontclient/src/Router/mainRouter.js:14]: oh thanks, I forgot to remove it.

**Comment by KombangkoeDias** [comment]
> I also encountered this warning in the console after I had logged in.
> 
> firefox_2021-03-28_20-17-19
> 
> I don't know whether it is severe or not, but it might be worth investigating.
> 
> Edit: I logged in again, and it was gone. No error. (o_o)

I got it too, somehow it seems to come with the delay router guard (waiting for the authentication data to come before validate) and the component got rendered before the data comes in and then got updated  or exited due to the data, that's why we need the Loading Component, hope it fixes this.

**Comment by KombangkoeDias** [comment]
fixed the import, now running fine. Will merge this tomorrow.

**Review by Megacodexz — APPROVED** [review]

**Pull request merged**